db/backup/recovery_test.py

```diff
@@ -0,0 +1,246 @@
+            print('Performing backup')
+        # Perform backup
+        subprocess.run(
+            [
+                'sudo',
+                'mariabackup',
+                '--copy-back',
+                f'--target-dir={LOCAL_BACKUP_FOLDER}/{backup_folder}',
+            ],
+            check=True,
+        )
+        # Fix permissions
+        subprocess.run(
+            ['sudo', 'chown', '-R', 'mysql:mysql', '/var/lib'],
+            check=True,
+        )
+        # Start the MariaDB server
+        subprocess.run(['sudo', 'systemctl', 'start', 'mariadb'], check=True)
+
+    def setUp(self):
+        self.prod_connection = ConnectionDetails('10.152.0.2', 'backup')
+        self.local_connection = ConnectionDetails('localhost', 'root')
+        backup_folder = pull_latest_backup()
+        self.timestamp = get_timestamp(backup_folder)
+
+    def test_database_exists(self):
+        """Validates that the db in the production
+        database matches those produced by the restored db"""
+        backup_databases = get_databases(self.local_connection)
+        prod_databases = get_databases(self.prod_connection)
+        self.assertEqual(backup_databases, prod_databases)
+
+    def test_row_number(self):
+        """ Tests that the count of rows for each table matches """
+        tables = get_results('show tables;', self.local_connection)
+        for table in tables:
+            if table[0] == 'DATABASECHANGELOG' or table[0] == 'DATABASECHANGELOGLOCK':
+                continue
+            restored_count = get_results(
+                f'SELECT COUNT(*) FROM {table[0]};', self.local_connection
+            )
+            prod_count = get_results(
+                f"SELECT COUNT(*) FROM {table[0]} FOR SYSTEM_TIME AS OF TIMESTAMP'{self.timestamp}'",
+                self.prod_connection,
+            )
+
+            self.assertEqual(restored_count, prod_count)
+
+    def test_rows_top(self):
+        """ Validates the top 10 rows in each table match """
+        tables = get_results('show tables;', self.local_connection)
+        for table in tables:
+            if table[0] == 'DATABASECHANGELOG' or table[0] == 'DATABASECHANGELOGLOCK':
+                continue
+            restored_results = get_results(
+                f'SELECT * FROM {table[0]} LIMIT 10;', self.local_connection
+            )
+            prod_results = get_results(
+                f"SELECT * FROM {table[0]} FOR SYSTEM_TIME AS OF TIMESTAMP'{self.timestamp}' LIMIT 10;",
+                self.prod_connection,
+            )
+
+            self.assertEqual(restored_results, prod_results)
+
+    def test_random_rows_ids(self):
+        """Pulls 10 random rows for testing.
+        Operates on tables with a unique id field"""
+
+        # Tables with unique id fields.
+        tables = [
+            'analysis',
+            'family',
+            'participant',
+            'project ',
+            'sample',
+            'sample_sequencing',
+        ]
```

Comment:
This now has a unique ID field:
```suggestion
            'family_participant',
        ]
```

Reply:
Awesome ty